ipmininet/ipnet.py

```diff
@@ -549,6 +550,64 @@ def ping6Pair(self):
            return: ploss packet loss percentage"""
         return self.pingPair(use_v4=False)
 
+   
+    def runFailurePlan(self,failure_plan):
+        """this function run a failure plan
+            the plan have to be a list of tuple with the name of routeur between
+            which the link have to be down
+            example of plan:
+            [("R1","R2"),("R8","R9"),("R3","R1")]
+            This will return a list of IPIntf 
+        """
+        print("** Starting failure plan")
+        interfaces_down = []
+        for link in failure_plan:
+            try:
+                node1 = self.get(link[0])
+                node2 = self.get(link[1])
+                interfaces = node1.connectionsTo(node2)
+                interfaceNode1 = interfaces[0][0]
+                interfaceNode2 = interfaces[0][1]
+                commande = "ip link set dev " + str(interfaceNode1) + " down"
+                node1.cmd(commande)
+                interfaces_down.append(interfaceNode1)
+                print("** link between " + link[0] + " and "+ link[1] + " down")
+            except Exception as e:
```

Comment:
You could be more specific, it is a KeyError

Reply:
Moreover this does not solve the comment of @oliviertilmans 

> This can throw an exception if the node name does not exists, which at the moment abruptly terminates everything.
You need to handle this locally here, and likely perform another round of validations in the CLI handler (so your failure plan is a transaction, i.e., all failures are applied or none are).

If an exception is triggered here, you will still have links that are shutdown and links that are not.
You need to first check that interfaces and links exist before executing the commands to remove them